sso/provider-keycloak/src/main/java/org/eclipse/kapua/sso/provider/keycloak/jwt/KeycloakJwtProcessor.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Eurotech and/or its affiliates and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua.sso.provider.keycloak.jwt;
+
+import org.eclipse.kapua.sso.exception.SsoJwtException;
+import org.eclipse.kapua.sso.provider.jwt.AbstractJwtProcessor;
+import org.eclipse.kapua.sso.provider.keycloak.KeycloakSingleSignOnUtils;
+import org.eclipse.kapua.sso.provider.keycloak.setting.KeycloakSsoSetting;
+import org.eclipse.kapua.sso.provider.keycloak.setting.KeycloakSsoSettingKeys;
+import org.eclipse.kapua.sso.provider.setting.SsoSetting;
+import org.eclipse.kapua.sso.provider.setting.SsoSettingKeys;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The Keycloak JWT Processor.
+ */
+public class KeycloakJwtProcessor extends AbstractJwtProcessor {
+
+    private static final SsoSetting SSO_SETTING = SsoSetting.getInstance();
+
+    public KeycloakJwtProcessor() throws SsoJwtException {
+        super(Duration.ofHours(1));
```

Comment:
`Duration.ofHours(1)` what makes last/take 1 hour?

Reply:
Actually it was there also in the old implementation, the old JwtProcessor constructor was called with `Duration.ofHours(1)` as parameter, see [here](https://github.com/eclipse/kapua/blob/2e00fdecb35f8ee6fa269baed97a8dfea173a2f7/service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/utils/JwtProcessors.java#L31). Do you have any suggestions to get rid of that?

Reply:
I was wondering what's used for and it seems related to the validy of the JWT token. 
Can we make this configurable?

Reply:
Yes, now this is configurable through the property `sso.openid.jwt-processor-timeout`